### ISSUE TITLE
`predict.dsmodel` not exported from namespace in 1.0.8

### ISSUE THREAD

**Issue by erex** [DistanceDevelopment/Distance#171]
Email from a user:

> I want to use it to obtain probability of detection for each of my sightings. However, the predict.dsmodel looks like it is not available in the newest version of the ds package. Are there other functions or packages I can use to obtain these probabilities?

Checking via an MRE, I find this is true, so offered these alternatives:

> But it is easy to remedy. You can either use the "triple colon" trick to access functions within a package that are not exported. Or you can simply call the generic function predict​ that checks what type of model object it has received and call the appropriate function as a result.  Here is an example of both (showing they produce identical results) using the amakihi​ dataset because I wanted a covariate in my detection function model so predict is doing something interesting:

```
library(Distance)
data("amakihi")
cu <- convert_units("meter", NULL, "square kilometer")
fitobs <- ds(amakihi, transect = "point", key="hr",
             truncation = 83, formula=~"OBs")
mypredict <- Distance:::predict.dsmodel(fitobs, compute=TRUE)
mypred2 <- predict(fitobs, compute = TRUE)
identical(mypredict, mypred2)
table(mypredict)
```

I presume the function in question should be exported in the next release.

**Comment by lenthomas** [comment]
I'm no expert, but it's not clear to me that S3 methods should be in the namespace.  See, for example [this stackoverflow post from some years ago](https://stackoverflow.com/questions/18512528/how-to-export-s3-method-so-it-is-available-in-namespace).  I expect there's more recent and definitive advice somewhere, perhaps in the `roxygen2` documentation -- @LHMarshall, do you have an opinion here?

**Comment by erex** [comment]
Perhaps the confusion on the part of the user is derived from the existence of help for the function `predict.dsmodel`. By looking at the list of functions in the package, they thought they should be able to call that function
![image](https://github.com/DistanceDevelopment/Distance/assets/330744/f98b9774-9ce5-41cd-8eef-2a0a5fae4bb6)
I don't know how to alleviate the confusion.

**Comment by lenthomas** [comment]
Yes, seems like a reasonable expectation that a function documented in the help for a package is exported, unless noted otherwise.  Let's see if @LHMarshall  knows what best practice is around this general issue and the predict method in particular (as well as other "generic" methods associated with objects such as the summary method) otherwise I'll look into it.

**Comment by LHMarshall** [comment]
I'm confused by this, it looks like this method is exported

```
> library(Distance)
> data("amakihi")
> cu <- convert_units("meter", NULL, "square kilometer")
> fitobs <- ds(amakihi, transect = "point", key="hr",
+              truncation = 83, formula=~"OBs")
Model contains covariate term(s): no adjustment terms will be included.
Fitting hazard-rate key function
AIC= 10787.754
> mypredict <- Distance:::predict.dsmodel(fitobs, compute=TRUE)
> mypred2 <- predict(fitobs, compute = TRUE)
> identical(mypredict, mypred2)
[1] TRUE
> table(mypredict)
fitted
0.180411590172534 0.206725067866751 0.386629089289073 
              208                65               970
```

Also checking the NAMESPACE I can see the following declaration

```
S3method(predict,dsmodel)
```

According to github the predict.dsmodel.R file has not been modified in 4 years so not sure why anything changed

https://github.com/DistanceDevelopment/Distance/blob/master/R/predict.dsmodel.R

This was first exported in 2020.

I can't see that this was changed in NAMESPACE either - I wondered if roxygen had got mixed up.
